lxd/device/cdi/configure.go

```diff
@@ -0,0 +1,335 @@
+package cdi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"tags.cncf.io/container-device-interface/specs-go"
+
+	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// specDevToNativeDev builds a list of unix-char devices to be created from a CDI spec.
+func specDevToNativeDev(configDevices *ConfigDevices, d specs.DeviceNode) error {
+	if d.Path == "" {
+		return fmt.Errorf("device path is empty in the CDI device node: %v", d)
+	}
+
+	var hostPath string
```

Comment:
Please can you add a comment explaining what is happening here? Why do only some devices have a HostPath?

Reply:
Also this is can be written just as:

```go
hostPath = d.HostPath
if hostPath == "" {
    hostPath = d.Path  // When the hostPath is empty, the path is the device path in the container.
}
```

Reply:
> Please can you add a comment explaining what is happening here? Why do only some devices have a HostPath?

`HostPath` can be omitted if it is the host path and dest path are the same in the spec.

Reply:
Right so that sounds like a good comment to add :)

Reply:
fixed